daml-assistant/daml-helper/src/DA/Daml/Helper/Ledger.hs

```diff
@@ -431,56 +429,58 @@ allocateParty args name = do
-          activeContracts <-
-            L.getActiveContracts
-              ledgerId
-              (TransactionFilter $
-               Map.fromList [(L.unParty p, Just $ Filters Nothing) | p <- parties])
-              (L.Verbosity False)
-          let chunks = chunksOf 100 activeContracts
-          forM_ chunks $ \chunk -> do
-            let cmds cmdId =
-                  L.Commands
-                    { coms =
-                        [ L.ExerciseCommand
-                          { tid = tid
-                          , cid = cid
-                          , choice = L.Choice "Archive"
-                          , arg = L.VRecord $ L.Record Nothing []
-                          }
-                        | (_offset, _mbWid, events) <- chunk
-                        , L.CreatedEvent {cid, tid} <- events
-                        ]
-                    , lid = ledgerId
-                    , wid = Nothing
-                    , aid = L.ApplicationId "ledger-reset"
-                    , cid = L.CommandId $ TL.fromStrict $ UUID.toText cmdId
-                    , actAs = parties
-                    , readAs = []
-                    , dedupPeriod = Nothing
-                    , minLeTimeAbs = Nothing
-                    , minLeTimeRel = Nothing
-                    , sid = Nothing
-                    }
-            let noCommands = null [ x | (_offset, _mbWid, events) <- chunk, x <- events ]
-            unless noCommands $ do
-              cmdId <- liftIO UUID.nextRandom
-              errOrEmpty <- L.submit $ cmds cmdId
-              case errOrEmpty of
-                Left err -> liftIO $ putStrLn $ "Failed to archive active contracts: " <> err
-                Right () -> pure ()
+-- TODO[SW] Implementation not ported to ledger-api-v1, as it wasn't fully working in the first place, and we're moving away from
+-- hs-bindings stream support. We may add this back later in a different form.
+-- runLedgerReset :: LedgerFlags -> IO ()
+-- runLedgerReset flags = do
+--   putStrLn "Resetting ledger."
+--   args <- getDefaultArgs flags
+--   reset args
+
+-- reset :: LedgerArgs -> IO ()
+-- reset args = do
+--     runWithLedgerArgs args $ do
+--         parties <- map L.party <$> L.listKnownParties
+--         unless (null parties) $ do
+--           ledgerId <- L.getLedgerIdentity
+--           activeContracts <-
+--             L.getActiveContracts
+--               ledgerId
+--               (TransactionFilter $
+--                Map.fromList [(L.unParty p, Just $ Filters Nothing) | p <- parties])
+--               (L.Verbosity False)
+--           let chunks = chunksOf 100 activeContracts
+--           forM_ chunks $ \chunk -> do
+--             let cmds cmdId =
+--                   L.Commands
+--                     { coms =
+--                         [ L.ExerciseCommand
+--                           { tid = tid
+--                           , cid = cid
+--                           , choice = L.Choice "Archive"
```

Comment:
So the reset was just an archive of all contracts? Hmm. I guess it helps in some tests and doesn't in other. We still are left with old parties and packages

Reply:
Exactly, it does unfortunately break an existing test though, so we do need to do something about this loss.
We're considering moving the hs-bindings functionality that we do use over to a service in scala, so we can fix this at that point.